experiments/state/src/main/scala/org/apache/gearpump/streaming/state/user/example/processor/WindowAverageProcessor.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.streaming.state.user.example.processor
+
+import com.twitter.algebird.{AveragedGroup, AveragedValue}
+import org.apache.gearpump.Message
+import org.apache.gearpump.cluster.UserConfig
+import org.apache.gearpump.streaming.state.system.impl.PersistentStateConfig
+import org.apache.gearpump.streaming.state.user.api.PersistentTask
+import org.apache.gearpump.streaming.task.{StartTime, TaskContext}
+import org.apache.gearpump.util.LogUtil
+import org.slf4j.Logger
+
+object WindowAverageProcessor {
+  val LOG: Logger = LogUtil.getLogger(classOf[WindowAverageProcessor])
+}
+
+class WindowAverageProcessor(taskContext : TaskContext, conf: UserConfig)
+  extends PersistentTask[AveragedValue](taskContext, conf) {
+
+  private val windowSize = conf.getLong(PersistentStateConfig.STATE_WINDOW_SIZE)
+    .getOrElse(throw new RuntimeException("state window size not configured"))
```

Comment:
We should never throw RuntimeException in actor.

Exception should also be avoided.